sdk/identity/identity/src/client/errors.ts

```diff
@@ -73,33 +87,41 @@ export class AuthenticationError extends Error {
   constructor(statusCode: number, errorBody: object | string | undefined | null) {
     let errorResponse = {
       error: "unknown",
-      error_description: "An unknown error occurred and no additional details are available."
+      errorDescription: "An unknown error occurred and no additional details are available."
     };
 
     if (isErrorResponse(errorBody)) {
-      errorResponse = errorBody;
+      errorResponse = {
+        error: errorBody.error,
+        errorDescription: errorBody.error_description
+      };      
     } else if (typeof errorBody === "string") {
       try {
         // Most error responses will contain JSON-formatted error details
-        // in the response body
-        errorResponse = JSON.parse(errorBody);
+        // in the response body        
+        const oauthErrorResponse: OAuthErrorResponse = JSON.parse(errorBody);
+
+        errorResponse = {
+          error: oauthErrorResponse.error,
+          errorDescription: oauthErrorResponse.error_description
```

Comment:
Can you also copy over the other fields from OAuthErrorResponse?

Reply:
Done. Also, added in a test so I can't mess that up again.